src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -875,6 +877,58 @@ internal ImmutableArray<LocalInfo<TypeSymbol>> DecodeLocalSignatureOrThrow(ref B
             }
         }
 
+        internal TypeSymbol DecodeGenericParameterConstraint(EntityHandle token, out bool isUnmanagedConstraint)
+        {
+            isUnmanagedConstraint = false;
+
+            switch (token.Kind)
+            {
+                case HandleKind.TypeSpecification:
+                    {
+                        try
+                        {
+                            var memoryReader = this.Module.GetTypeSpecificationSignatureReaderOrThrow((TypeSpecificationHandle)token);
+                            var modifiers = DecodeModifiersOrThrow(ref memoryReader, AllowedRequiredModifierType.System_Runtime_InteropServices_UnmanagedType, out var typeCode, out var modReqFound);
+                            var type = DecodeTypeOrThrow(ref memoryReader, typeCode, out _);
+
+                            if (modReqFound)
+                            {
+                                if (type.SpecialType == SpecialType.System_ValueType)
+                                {
+                                    isUnmanagedConstraint = true;
+                                }
+                                else
+                                {
+                                    return GetUnsupportedMetadataTypeSymbol();
+                                }
+                            }
+                            else if (!modifiers.IsDefaultOrEmpty)
+                            {
+                                // Optional modifiers (also, other required parameters) not allowed on generic parameters
```

Comment:
>parameters [](start = 76, length = 10)

What parameters does this refer to?

Reply:
Bad phrase. Will fix.

---
In reply to: [175516830](https://github.com/dotnet/roslyn/pull/24817#discussion_r175516830) [](ancestors = 175516830)